source/common/runtime/runtime_impl.h

```diff
@@ -70,15 +74,29 @@ class SnapshotImpl : public Snapshot, public ThreadLocal::ThreadLocalObject {
   bool featureEnabled(const std::string& key, uint64_t default_value) const override;
   bool featureEnabled(const std::string& key, uint64_t default_value,
                       uint64_t random_value) const override;
+  bool featureEnabled(const std::string& key,
+                      const envoy::type::FractionalPercent& default_value) const override;
+  bool featureEnabled(const std::string& key, const envoy::type::FractionalPercent& default_value,
+                      uint64_t random_value) const override;
   const std::string& get(const std::string& key) const override;
   uint64_t getInteger(const std::string& key, uint64_t default_value) const override;
   const std::vector<OverrideLayerConstPtr>& getLayers() const override;
 
   static Entry createEntry(const std::string& value);
 
 private:
+  static void resolveEntryType(Entry& entry) {
+    if (parseEntryUintValue(entry)) {
+      return;
+    }
+    parseEntryFractionalPercentValue(entry);
+  }
+
+  static bool parseEntryUintValue(Entry& entry);
+  static bool parseEntryFractionalPercentValue(Entry& entry);
```

Comment:
nit: I don't believe the return value here is used so I would make it void for now.